elide-core/src/main/java/com/yahoo/elide/security/checks/ExtractedChecks.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.security.checks;
+
+import com.yahoo.elide.audit.InvalidSyntaxException;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.PersistentResource;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Extracted checks.
+ */
+@Slf4j
+public class ExtractedChecks {
+    private final Class<? extends Check>[] anyEntityChecks;
+    private final Class<? extends Check>[] allEntityChecks;
+    private final Class<? extends Check>[] anyFieldChecks;
+    private final Class<? extends Check>[] allFieldChecks;
+
+    /**
+     * Constructor.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation class
+     * @param <A> Type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final PersistentResource resource, final Class<A> annotationClass) {
+        this(resource, annotationClass, null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation class
+     * @param field Field
+     * @param <A> Type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final PersistentResource resource,
+                                                  final Class<A> annotationClass,
+                                                  final String field) {
+        this(resource.getResourceClass(), resource.getDictionary(), annotationClass, field);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param cls Entity class
+     * @param dictionary Entity dictionary
+     * @param annotationClass annotation class
+     * @param field Field
+     * @param <A> type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final Class<?> cls,
+                                                  final EntityDictionary dictionary,
+                                                  final Class<A> annotationClass,
+                                                  final String field) {
+        final A entityAnnotation = dictionary.getAnnotation(cls, annotationClass);
+        final A fieldAnnotation = (field == null) ? null
+                                                  : dictionary.getAttributeOrRelationAnnotation(
+                                                        cls,
+                                                        annotationClass,
+                                                        field);
+        // No checks specified
+        if (entityAnnotation == null) {
+            anyEntityChecks = null;
+            allEntityChecks = null;
+        } else {
```

Comment:
This else blocks looks 'unrolled'.  Maybe another function here?  It might also help with cyclomatic complexity.